components/prism-factor.js

```diff
@@ -0,0 +1,277 @@
+				The values of MMMMMMMMMMMMM and EEEE map directly to the mantissa and exponent fields of the binary IEEE 754 representation."
+				<https://docs.factorcode.org/content/article-syntax-floats.html>
+			*/
+			{ 	pattern: /(^|\s)[+-]?0(?:b1\.[01]*|o1\.[0-7]*|d1\.\d*|x1\.[\dA-F]*)p\d+(?=\s|$)/i,
+				lookbehind: true }
+		],
+
+		boolean: {
+			pattern: /(^|\s|\r\n)(?:t|f)(?=\s|$)/,
+			lookbehind: true
+		},
+
+		'special-string-literal': {
+			pattern: /(^|\s)[A-Z0-9\-]+"\s(?:\\"|[^"])*"/,
+			lookbehind: true,
+			greedy: true,
+			alias: 'string'
+		},
+
+		'special-using': {
+			pattern: /(^|\s)USING:(?:\s\S+)*(?=\s+;(?:\s|$))/,
+			lookbehind: true,
+			alias: 'function',
+			inside: {
+				// this is essentially a regex for vocab names, which i don't want to specify
+				// but the USING: gets picked up as a vocab name
+				string: {
+					pattern: /(\s)[^:\r\n\t ]+/,
+					lookbehind: true
+				}
+			}
+		},
+
+		/* this description of stack effect literal syntax is not complete and not as specific as theoretically possible
+		 	trying to do better is more work and regex-computation-time than it's worth though.
+			- we'd like to have the "delimiter" parts of the stack effect [ (, --, and ) ] be a different (less-important or comment-like) colour to the stack effect contents
+			- we'd like if nested stack effects were treated as such rather than just appearing flat
+			- we'd like if the following variable name conventions were recognised specifically:
+				special row variables = ..a b..
+				nested quotation variables end with a colon = quot: ( a: ( -- ) -- b )
+				word throws unconditional error = *
+				any other word-like variable name = a ? q' etc
+
+			https://docs.factorcode.org/content/article-effects.html
+
+			these are pretty complicated to highlight properly without a real parser, and therefore out of scope
+		'stack-effect-old': {
+			pattern: /(^|\s)(?:call|execute|eval)?\((?:\s+[^"\r\n\t ]\S*)*?\s+--(?:\s+[^"\n\t ]\S*)*?\s+\)(?=\s|$)/,
+			lookbehind: true,
+			alias: 'operator',
+		},
+
+		TODO:
+		'stack-effect-variable': {
+			{ pattern: / /, lookbehind: true, alias: 'operator' },
+		},
+		*/
+
+		// current solution is not great
+		'stack-effect-open': {
+			pattern: /(^|\s)(?:call|execute|eval)?\((?=\s)/,
+			lookbehind: true,
+			alias: 'operator' },
+		'stack-effect-middle': {
+			pattern: /(\s)--(?=\s)/,
+			lookbehind: true,
+			alias: 'operator' },
+		'stack-effect-close': {
+			pattern: /(\s)\)(?=\s|$)/,
+			lookbehind: true,
+			alias: 'operator' },
+
+		combinators: [],
+
+		'kernel-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'sequences-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'math-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'constructor-word': {
+			// <array> but not <=>
+			pattern: /(^|\s)<(?!=+>|-+>)[^\r\n\t ]+>(?=\s|$)/,
```

Comment:
`\f` and `\v` (which are in `\s`) may appear literally in a word name, as they are not treated as whitespace. A literal tab character is forbidden from appearing anywhere in a program. 

`\s` and `\S` to quantify whitespace in tokenising Factor is wrong compared to the reference implementation, but using `(?:\n|\r\n)` everywhere (which would be perfectly correct) is a pain. I've tried to allow `\f` and `\v` in word names where possible, to align with the reference implementation.

Reply:
I also thought that JavaScript disallowed `[^\s]`, i.e. a builtin class other than an escaped literal inside a negative class, but now I see it's allowed, I might as well change them...

Reply:
I don't see why any language would allow `\f` and `\v` in names... An interesting choice on factor's side. 

I can't believe people actually use this. Seems to be part of the technically-possible-but-please-don't category.

Reply:
I don't know why it's like this, and indeed nobody uses non-printing characters in word names, but the reference implementation recognises only  the space character as horizontal whitespace, and LF/CRLF as vertical whitespace.